src/lavinmq/auth/authenticator.cr

```diff
@@ -0,0 +1,9 @@
+module LavinMQ
+  module Auth
+    abstract class Authenticator
+      Log = LavinMQ::Log.for "auth.handler"
+
+      abstract def authenticate(username : String, password : String)
```

Comment:
Maybe it would be nice to be explicit about the return type here
```suggestion
      abstract def authenticate(username : String, password : String) : User?
```